src/irr/asset/CGraphicsPipelineLoaderMTL.cpp

```diff
@@ -0,0 +1,467 @@
+#include "irr/asset/CGraphicsPipelineLoaderMTL.h"
+
+using namespace irr;
+using namespace asset;
+
+bool CGraphicsPipelineLoaderMTL::isALoadableFileFormat(io::IReadFile* _file) const
+{
+    if (!_file)
+        return false;
+
+    const size_t prevPos = _file->getPos();
+
+    _file->seek(0ull);
+
+    std::string mtl;
+    mtl.resize(_file->getSize());
+    _file->read(mtl.data(), _file->getSize());
+    _file->seek(prevPos);
+
+    return mtl.find("newmtl") != std::string::npos;
+}
+
+namespace
+{
+    core::smart_refctd_ptr<ICPUPipelineLayout> makePipelineLayoutFromMtl(const CMTLPipelineMetadata::SMtl& _mtl)
+    {
+        const size_t textureCnt = std::count_if(_mtl.maps, _mtl.maps + CMTLPipelineMetadata::SMtl::EMP_REFL_POSX + 1u, [](const std::string& _path) { return !_path.empty(); });
+        auto bindings = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<ICPUDescriptorSetLayout::SBinding>>(textureCnt);
```

Comment:
cool but you need to add one more binding for a per-material UBO (for materials parameters)

Reply:
and another set 1 UBO for MVP matrix, and MV inverse transpose, etc.

Plus possibly push constant for the illumination model etc.

Reply:
i figured, loader only creates sets with descriptors provided by it. However i think putting layout for set=1 sounds like a good idea to me. Anyway.. should the buffer (the one being the UBO asset in the set) be filled wih ny particular data?

Reply:
well if the shader requires stuff that is "constant" (not world matrix) like stuff that you'd have to fill from the metadata to make the material "work", like illumination model, ambient and diffuse, etc. then you should sure create, attach and fill an UBO or fill a push constant with data.